Source/Particles/Collision/PairWiseCoulombCollision.cpp

```diff
@@ -1,56 +1,71 @@
-/* Copyright 2019 Yinjian Zhao
+/* Copyright 2020 Yinjian Zhao, David Grote
  *
  * This file is part of WarpX.
  *
  * License: BSD-3-Clause-LBNL
  */
-#include "CollisionType.H"
+#include "PairWiseCoulombCollision.H"
 #include "ShuffleFisherYates.H"
 #include "ElasticCollisionPerez.H"
 #include "Utils/ParticleUtils.H"
 #include "Utils/WarpXUtil.H"
 #include "WarpX.H"
 
-CollisionType::CollisionType(
-    const std::vector<std::string>& species_names,
-    std::string const collision_name)
+using namespace amrex::literals;
+
+PairWiseCoulombCollision::PairWiseCoulombCollision (std::string const collision_name)
+    : CollisionBase(collision_name)
 {
 
-    // read collision species
-    std::vector<std::string> collision_species;
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(m_species_names.size() == 2,
+                                     "Pair wise Coulomb must have exactly two species.");
+
     amrex::ParmParse pp(collision_name);
-    pp.getarr("species", collision_species);
-    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(collision_species.size() == 2,
-    "Collision species must name exactly two species.");
 
     // default Coulomb log, if < 0, will be computed automatically
-    m_CoulombLog = -1.0;
+    m_CoulombLog = -1.0_rt;
     queryWithParser(pp, "CoulombLog", m_CoulombLog);
 
-    // number of time steps between collisions
-    m_ndt = 1;
-    pp.query("ndt", m_ndt);
-
-    for (int i=0; i<static_cast<int>(species_names.size()); i++)
-    {
-        if (species_names[i] == collision_species[0])
-        { m_species1_index = i; }
-        if (species_names[i] == collision_species[1])
-        { m_species2_index = i; }
-    }
-
-    if (collision_species[0] == collision_species[1])
+    if (m_species_names[0] == m_species_names[1])
         m_isSameSpecies = true;
     else
         m_isSameSpecies = false;
 
 }
 
-using namespace amrex;
+void
+PairWiseCoulombCollision::doCollisions (amrex::Real cur_time, MultiParticleContainer* mypc)
+{
+
+    const amrex::Real dt = WarpX::GetInstance().getdt(0);
+    if ( int(std::floor(cur_time/dt)) % m_ndt != 0 ) return;
+
+    auto& species1 = mypc->GetParticleContainerFromName(m_species_names[0]);
+    auto& species2 = mypc->GetParticleContainerFromName(m_species_names[1]);
+
+    // Enable tiling
+    amrex::MFItInfo info;
+    if (amrex::Gpu::notInLaunchRegion()) info.EnableTiling(species1.tile_size);
+
+    // Loop over refinement levels
+    for (int lev = 0; lev <= species1.finestLevel(); ++lev){
+
+        // Loop over all grids/tiles at this level
+#ifdef _OPENMP
+        info.SetDynamic(true);
```

Comment:
I know it's in the original code, but I seem to remember that we set this flag as a general input option these days and not on a per-kernel basis anymore.

Reply:
This is somewhat tangential to this PR so the fix should go into a separate mini-PR.